public/components/event_analytics/explorer/sidebar/sidebar.tsx

```diff
@@ -89,22 +93,34 @@ export const Sidebar = (props: ISidebarProps) => {
 
   const handleAddField = useCallback(
     (field: IField) => {
-      updateStoreFields(
-        toggleFields(explorerFields, field, AVAILABLE_FIELDS, SELECTED_FIELDS),
-        tabId,
-        SELECTED_FIELDS
-      );
+      let nextFields;
+      if (
+        explorerFields.selectedFields.length === 0 &&
+        storedExplorerFields.selectedFields.length !== 0
+      ) {
+        nextFields = toggleFields(storedExplorerFields, field, AVAILABLE_FIELDS, SELECTED_FIELDS);
+      } else {
+        nextFields = toggleFields(explorerFields, field, AVAILABLE_FIELDS, SELECTED_FIELDS);
+      }
+      updateStoreFields(nextFields, tabId, SELECTED_FIELDS);
+      setStoredExplorerFields(nextFields);
     },
     [explorerFields, tabId]
   );
 
   const handleRemoveField = useCallback(
     (field: IField) => {
-      updateStoreFields(
-        toggleFields(explorerFields, field, SELECTED_FIELDS, AVAILABLE_FIELDS),
-        tabId,
-        AVAILABLE_FIELDS
-      );
+      let nextFields;
+      if (
+        explorerFields.selectedFields.length === 0 &&
+        storedExplorerFields.selectedFields.length !== 0
+      ) {
+        nextFields = toggleFields(storedExplorerFields, field, SELECTED_FIELDS, AVAILABLE_FIELDS);
+      } else {
+        nextFields = toggleFields(explorerFields, field, SELECTED_FIELDS, AVAILABLE_FIELDS);
+      }
```

Comment:
Minor: Can this be simplified? Suggesting something below (can be better)
```
    const selectedFields =
      explorerFields.selectedFields.length === 0 && storedExplorerFields.selectedFields.length > 0
        ? storedExplorerFields
        : explorerFields;

    const nextFields = toggleFields(selectedFields, field, SELECTED_FIELDS, AVAILABLE_FIELDS);

Reply:
Also I see something similar in `handleAddField` do you want to create a function and reuse code?

Reply:
Yep, will make the change now